src/wad/visitor.rs

```diff
@@ -159,65 +170,79 @@ impl<'a, V: LevelVisitor> LevelWalker<'a, V> {
             self.seg(sector, seg, (v1, v2));
         }
 
-        // The convex polyon defined at the intersection of the partition lines, intersected with
+        // The convex polyon defined at the intersection of the partition lines,
+        // intersected with
         // the half-volumes of the segs form the 'implicit' points.
         for i_line in 0..(self.bsp_lines.len() - 1) {
             for j_line in (i_line + 1)..self.bsp_lines.len() {
                 let (l1, l2) = (&self.bsp_lines[i_line], &self.bsp_lines[j_line]);
                 let point = match l1.intersect_point(l2) {
                     Some(p) => p,
-                    None => continue
+                    None => continue,
                 };
 
                 let dist = |l: &Line2f| l.signed_distance(&point);
 
                 // The intersection point must lie both within the BSP volume
                 // and the segs volume.
-                if self.bsp_lines.iter().map(|x| dist(x)).all(|d| d >= -BSP_TOLERANCE)
-                        && self.subsector_seg_lines.iter().map(dist).all(|d| d <= SEG_TOLERANCE) {
+                if self.bsp_lines.iter().map(|x| dist(x)).all(|d| d >= -BSP_TOLERANCE) &&
+                   self.subsector_seg_lines.iter().map(dist).all(|d| d <= SEG_TOLERANCE) {
                     self.subsector_points.push(point);
                 }
             }
         }
         if self.subsector_points.len() < 3 {
-            warn!("Degenerate source polygon {} ({} vertices).", id, self.subsector_points.len());
+            warn!("Degenerate source polygon {} ({} vertices).",
+                  id,
+                  self.subsector_points.len());
         }
         points_to_polygon(&mut self.subsector_points);  // Sort and remove duplicates.
         if self.subsector_points.len() < 3 {
             warn!("Degenerate cannonicalised polygon {} ({} vertices).",
-                  id, self.subsector_points.len());
+                  id,
+                  self.subsector_points.len());
         } else {
             self.flat_poly(sector);
         }
     }
 
     fn seg(&mut self, sector: &WadSector, seg: &WadSeg, vertices: (Vec2f, Vec2f)) {
-        let line = if let Some(line) = self.level.seg_linedef(seg) { line }
```

Comment:
I'm obviously a big fan of the `let else` proposal, but happy to leave it the way rustfmt wants it until that lands.